01_02_infrastructure_resources/infrastructure_resources.md

```diff
@@ -8,67 +8,484 @@ backgroundImage: url('../slides-support/common/4k-slide-bg-white.png')
+<div>
+
++ [spack.io](https://spack.io) is a package manager for supercomputers, Linux and macOS.
++ It makes it easy to install specific versions of scientific libraries that use a specific compiler, or architecture specific compilation flags.
++ can be installed as user without root privilegies
++ easy to package your own software with it
 
-foobar
+</div>
+<div>
+
+Getting started installing a library (libelf), as described in the [spack documentation](https://spack.readthedocs.io/en/latest/):
+```bash
+$ git clone -c feature.manyFiles=true https://github.com/spack/spack.git
+$ cd spack/bin
+$ ./spack install libelf
 ```
 
++ packaging your tool/library supports git or github as sources, which makes creating packages for your tool relatively easy as described in the [packaging guide](https://spack.readthedocs.io/en/latest/packaging_guide.html).
+
+</div>
+</div>
+
 ---
 
-# A Slide with two columns
+<div class="twocolumns">
+<div>
+
+# Conan
+
++ [conan.io]() is an open source, decentralized C/C++ package manager.
++ it is well integrated with buildtools (cmake, scons,...)
++ it is easy to [create packages](https://docs.conan.io/2/tutorial/creating_packages.html)
+
+</div>
+<div>
+
+`conanfile.txt` from the [conan tutorial](https://docs.conan.io/2/tutorial.html)
+```ini
+[requires]
+zlib/1.2.11
+[tool_requires]
+cmake/3.22.6
+[generators]
+CMakeDeps
+CMakeToolchain
+```
+
+and its companion `CMakeLists.txt`
+```cmake
+cmake_minimum_required(VERSION 3.15)
+project(compressor C)
+find_package(ZLIB REQUIRED)
+message("Building with CMake version: ${CMAKE_VERSION}")
+add_executable(${PROJECT_NAME} src/main.c)
+target_link_libraries(${PROJECT_NAME} ZLIB::ZLIB)
+```
+
+```bash
+cmake .. -DCMAKE_TOOLCHAIN_FILE=conan_toolchain.cmake \
+  -DCMAKE_BUILD_TYPE=Release
+```
+
+</div>
+</div>
+
+---
+# Vcpkg
 
 <div class="twocolumns">
 <div>
 
-## Code
++ [vcpkg.io](https://vcpkg.io/en/) is a free C/C++ package manager for acquiring and managing libraries
++ Maintained by the Microsoft C++ team and open source contributors.
+
+
+</div>
+<div>
 
-```c++
-void f(int x) {
-}
+```
+git clone https://github.com/Microsoft/vcpkg.git
+```
+Make sure you are in the directory you want the tool installed to before doing this.
+```bash
+./vcpkg/bootstrap-vcpkg.sh
+```
+Install libraries for your project
+```
+vcpkg install [packages to install]
+```
+Using vcpkg with CMake
+```
+cmake -B [build directory] -S . \
+-DCMAKE_TOOLCHAIN_FILE=[path to vcpkg]/scripts/buildsystems/vcpkg.cmake
 ```
 
 </div>
+</div>
+
+---
+# Web Resources
+
+* [godbolt.org](https://godbolt.org) (Compiler Explorer)
+    + *many* different compilers
+    + nicely annotated assembly output
+    + supports several popular libraries (ranges, {fmt}, …)
+    + code round-trip between Compiler Explorer, C++ Insights and Quick Bench possible
+    + we will use it quite a bit
+* [quick-bench.com](https://quick-bench.com) a site to easily a quickly benchmark C++ code (and inspect disassembly)
```

Comment:
Maybe worth adding the following in this section:
- https://gcc-explorer.com/
- https://build-bench.com
?